allensdk/test/brain_observatory/behavior/test_stimulus_processing.py

```diff
@@ -64,13 +148,118 @@ def test_get_draw_epochs(start_frame, stop_frame, expected):
 #     # convert_filepath_caseinsensitive prevents using any tempdirs/tempfiles
 
 
-# def test_get_stimulus_presentations():
-#     pass
-#     # TODO
-#     # Monster function with undocumented dependencies
+@pytest.mark.parametrize("behavior_stimuli_data_fixture, remove_stimuli, "
+                         "expected_metadata", [
+                             ({'grating_phase': 10.0,
+                               'grating_frequency': 90.0},
+                              ['images'],
+                              {'image_index': [0, 1, 2, 3, 4],
+                               'image_name': ['gratings_0.0', 'gratings_90.0',
```

Comment:
Can you please clarify why the gratings have a name here but not in your expected results for `get_stimuli_df`?

Reply:
When the grating metadata is merged, one level above this it merges on name of the stimuli, which for gratings are 'grating_0, grating_90, grating_180, grating_270'. Therefore the four types of gratings that can be present are defined with those names here to agree with the already existing merge logic one level above.